src/display/text_layer.js

```diff
@@ -213,19 +531,85 @@ var renderTextLayer = (function renderTextLayerClosure() {
       var styles = this._textContent.styles;
       var textDivs = this._textDivs;
       var viewport = this._viewport;
+      var bounds = [];
+      var enhanceTextSelection = this._enhanceTextSelection;
+      var enhanceOnClick = this._enhanceOnClick;
       for (var i = 0, len = textItems.length; i < len; i++) {
-        appendText(textDivs, viewport, textItems[i], styles);
+        appendText(textDivs, viewport, textItems[i], styles, bounds,
+                   enhanceTextSelection);
+      }
+      if (enhanceTextSelection) {
+        expand(bounds, this._viewport);
+        if (enhanceOnClick) {
+          this._bindMouse(textDivs);
+        }
       }
-
       if (!timeout) { // Render right away
-        render(this);
+        render(this, enhanceOnClick);
       } else { // Schedule
         var self = this;
         this._renderTimer = setTimeout(function() {
-          render(self);
+          render(self, enhanceOnClick);
           self._renderTimer = null;
         }, timeout);
       }
+    },
+
+    _bindMouse: function TextLayer_bindMouse(textDivs) {
+      for (var i in textDivs) {
+        textDivs[i].addEventListener('mousedown', this._onMousedown(textDivs));
+        textDivs[i].addEventListener('mouseup', this._onMouseup(textDivs));
+      }
+    },
+    _onMousedown: function TextLayer_onMousedown(textDivs) {
+      return function() {
+        for (var i in textDivs) {
```

Comment:
Nit: Since you know that `textDivs` is an Array, it's probably more efficient to instead do:
```
for (var i = 0, ii = textDivs.length; i < ii; i++) {
```
to iterate through the elements, given that `for..in` will unnecessarily stringify the indexes.